# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ModuleLoadRequest.h
# Commit: a71a2a98b6bc
# Full Hash: a71a2a98b6bc017a7bb60d4b7c41f543f156eede
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-21 21:39:46
# Regressor Bug: 1742437
# File Overlap Count: 1
# Description:
#   Bug 1742437 - move mElement from ScriptFetchOptions to DOMScriptLoadContext; r=jonco,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D136742
# ==============================================================================

diff -r d52858ae58d2 -r a71a2a98b6bc dom/script/ModuleLoadRequest.h
--- a/dom/script/ModuleLoadRequest.h	Mon Feb 21 15:39:04 2022 +0000
+++ b/dom/script/ModuleLoadRequest.h	Mon Feb 21 15:39:05 2022 +0000
@@ -51,11 +51,9 @@
                                                          ScriptLoadRequest)
 
   // Create a top-level module load request.
-  static ModuleLoadRequest* CreateTopLevel(nsIURI* aURI,
-                                           ScriptFetchOptions* aFetchOptions,
-                                           const SRIMetadata& aIntegrity,
-                                           nsIURI* aReferrer,
-                                           ScriptLoader* aLoader);
+  static ModuleLoadRequest* CreateTopLevel(
+      nsIURI* aURI, ScriptFetchOptions* aFetchOptions, Element* aElement,
+      const SRIMetadata& aIntegrity, nsIURI* aReferrer, ScriptLoader* aLoader);
 
   // Create a module load request for a static module import.
   static ModuleLoadRequest* CreateStaticImport(nsIURI* aURI,
@@ -64,7 +62,8 @@
   // Create a module load request for dynamic module import.
   static ModuleLoadRequest* CreateDynamicImport(
       nsIURI* aURI, ScriptFetchOptions* aFetchOptions, nsIURI* aBaseURL,
-      ScriptLoader* aLoader, JS::Handle<JS::Value> aReferencingPrivate,
+      Element* aElement, ScriptLoader* aLoader,
+      JS::Handle<JS::Value> aReferencingPrivate,
       JS::Handle<JSString*> aSpecifier, JS::Handle<JSObject*> aPromise);
 
   bool IsTopLevel() const override { return mIsTopLevel; }